src/types.internal.ts

```diff
@@ -11,3 +11,45 @@ export type CostMode = TupleToUnion<typeof CostModes>;
 
 export const SortOrders = ['desc', 'asc'] as const;
 export type SortOrder = TupleToUnion<typeof SortOrders>;
+
+export const SessionWindowSchema = v.object({
+	windowId: v.string(), // Format: YYYY-MM-DD-HH (start of 5-hour window)
+	startTime: v.string(), // ISO timestamp of first message in window
+	endTime: v.string(), // ISO timestamp of last message in window
+	inputTokens: v.number(),
+	outputTokens: v.number(),
+	cacheCreationTokens: v.number(),
+	cacheReadTokens: v.number(),
+	totalCost: v.number(),
+	messageCount: v.number(), // Number of messages in this window
+	conversationCount: v.number(), // Number of unique conversations in this window
+});
+
+export type SessionWindow = v.InferOutput<typeof SessionWindowSchema>;
+
+export const CurrentSessionInfoSchema = v.object({
+	hasActiveSession: v.boolean(),
+	timeRemainingMs: v.number(),
+	timeRemainingFormatted: v.string(),
+	activeWindow: v.optional(SessionWindowSchema),
+});
+
+export type CurrentSessionInfo = v.InferOutput<typeof CurrentSessionInfoSchema>;
+
+export const SessionWindowStatsSchema = v.object({
+	month: v.pipe(
+		v.string(),
+		v.regex(/^\d{4}-\d{2}$/), // YYYY-MM format
+	),
+	totalSessions: v.number(), // Total number of 5-hour windows with activity
+	remainingSessions: v.number(), // 50 - totalSessions (Max plan limit)
+	utilizationPercent: v.number(), // (totalSessions / 50) * 100
+	totalCost: v.number(),
+	totalTokens: v.number(),
+	averageCostPerSession: v.number(),
+	averageTokensPerSession: v.number(),
+	windows: v.array(SessionWindowSchema),
+	currentSession: v.optional(CurrentSessionInfoSchema),
+});
+
+export type SessionWindowStats = v.InferOutput<typeof SessionWindowStatsSchema>;
```

Comment:
you can define those types in `data-loader.ts` because we don't use it in other files.
also you can stop exporting it